### ISSUE TITLE
Consider alternatives to `theme.styles`

### ISSUE THREAD

**Issue by jxnblk** [system-ui/theme-ui#34]
Currently the `theme.styles` object is used as a shortcut to style components that are used in MDX. Each key in this object maps to a component that is passed to the MDXProvider's context. This allows the styles used in MDX to be generated and/or defined in multiple ways, including using Typography.js as an input. There might be alternative, more component-based approaches to handling this.

Currently, Theme UI passes components like the following to MDX context:

```js
// mdx components
{
  h1: styled('h1')(props => props.theme.styles.h1)
}
```

And the styles for this component can be defined in the `theme.styles` object like this:

```js
theme: {
  styles: {
    h1: {
      color: 'tomato',
    }
  }
}
```

**Comment by erikdstock** [comment]
I really like this construct overall but have a few minor comments:
- `theme.styles` is a specific, magic key, while others (`theme.textStyles` ) might be totally arbitrary
- It's a bit confusing how some styles might be specified here and others in a `variant`. I wonder if there could be some differentiation like `htmlStyles` and `componentStyles`
- The mdx use case is great but not necessary for every project:
  - if I'm building a design system for use across multiple projects, this interface makes it easy to style standard html elements - I can replace `<p>Text</p>` with `<Styled.p>Text</p>` and keep my html styles colocated with my theme (call it `styles`, `htmlStyles` whatever).
  - I might decide to extend this theme to react-native (haven't looked too deep into this possibility with theme-ui) and add a `reactNativeStyles` key.
  - In short it's likely that not all will use MDX, and it starts to feel a little cluttered to have the [nested provider](https://github.com/system-ui/theme-ui/blob/8e3087cc6b114dc13a054427f6de3c054b8091da/packages/theme-ui/src/provider.js#L54) automatically depend on `MDXProvider`. In short I'd ask if this could be moved to another package with an api roughly like:
```js
exports.wrapRootElement = ({element}) => (
  <ThemeProvider>
    <ThemeUIMDXProvider components={{ ...Styled, ...otherComponentsIWant }}>
      {element}
...)
```

**Comment by johno** [comment]
> This allows the styles used in MDX to be generated and/or defined in multiple ways, including using Typography.js as an input. There might be alternative, more component-based approaches to handling this.

In some ways I think it's beneficial to not be component based here so that the theme.js file and its `styles` are JSON serializable. When it boils down to it, the composable/component model already exists via theme/MDX context so it's something folks could already opt into.

Maybe you have other ideas on how a component-based model might work?
